### PR TITLE
Azure support

### PR DESCRIPTION
**Pull request by nishant-d**
fixes: devtron-labs/devtron#453

### CODE REVIEW

File: pkg/git/GitCliUtil.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+func NewGitUtil(logger *zap.SugaredLogger) *GitUtil {
+	return &GitUtil{
+		logger: logger,
+	}
+}
+
+const GIT_AKS_PASS = "/git-ask-pass.sh"
+
+func (impl *GitUtil) Fetch(rootDir string, username string, password string) (response, errMsg string, err error) {
```
Comment (pghildiyal) [inline pkg/git/GitCliUtil.go:25]: lfs support add kar dete hain

File: pkg/git/GitCliUtil.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	impl.logger.Debugw("git fetch ", "location", rootDir)
+	cmd := exec.Command("git", "-C", rootDir, "fetch", "origin", "--tags", "--force")
+	output, errMsg, err := impl.runCommandWithCred(cmd, username, password)
+	impl.logger.Debugw("fetch output", "root", rootDir, "opt", output, "errMsg", errMsg, "error", err)
+	return output, "", nil
+}
+
+func (impl *GitUtil) Checkout(rootDir string, branch string) (response, errMsg string, err error) {
+	impl.logger.Debugw("git fetch ", "location", rootDir)
```
Comment (pghildiyal) [inline pkg/git/GitCliUtil.go:34]: change fetch -> checkout

File: pkg/git/GitCliUtil.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	cmd := exec.Command("git", "-C", rootDir, "checkout", branch, "--force")
+	output, errMsg, err := impl.runCommand(cmd)
+	impl.logger.Debugw("fetch output", "root", rootDir, "opt", output, "errMsg", errMsg, "error", err)
```
Comment (pghildiyal) [inline pkg/git/GitCliUtil.go:37]: change fetch -> checkout

File: pkg/git/GitCliUtil.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	return output, "", nil
+}
+
+func (impl *GitUtil) runCommandWithCred(cmd *exec.Cmd, userName, password string) (response, errMsg string, err error) {
+	cmd.Env = append(os.Environ(),
```
Comment (pghildiyal) [inline pkg/git/GitCliUtil.go:42]: it will only be used for AKS? also will be ASKPASS or AKSPASS ?

File: pkg/git/RepositoryManager.go [hunk 5]
```diff
@@ -46,24 +46,31 @@ type RepositoryManager interface {
...
 
-func (impl RepositoryManagerImpl) Add(location string, url string, auth transport.AuthMethod) error {
+func (impl RepositoryManagerImpl) Add(location string, url string, userName, password string) error {
 	err := os.RemoveAll(location)
 	if err != nil {
 		impl.logger.Errorw("error in cleaning checkoutpath", "err", err)
 		return err
 	}
-	_, err = impl.clone(auth, location, url)
+	err = impl.gitUtil.Init(location, url, true)
```
Comment (pghildiyal) [inline pkg/git/RepositoryManager.go:63]: will this be called only once? repeated call may return error

File: pkg/git/GitCliUtil.go (hunk 4 above)
Comment (nishant-d) [reply pkg/git/GitCliUtil.go:42]: yas for ASK. variable is `ASKPASS`

File: pkg/git/RepositoryManager.go (hunk 5 above)
Comment (nishant-d) [reply pkg/git/RepositoryManager.go:63]: yes only once for fresh checkout

**Pull request merged**